main.c

```diff
@@ -0,0 +1,200 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "canlib/can.h"
+#include "canlib/can_common.h"
+#include "canlib/pic18f26k83/pic18f26k83_can.h"
+#include "canlib/message_types.h"
+#include "canlib/util/timing_util.h"
+#include "canlib/util/can_tx_buffer.h"
+#include "canlib/pic18f26k83/pic18f26k83_timer.h"
+
+#include "arming.h"
+#include "altitude_parsing.h"
+#include "mcc_generated_files/mcc.h"
+
+#define _XTAL_FREQ 12000000 //12MHz
+
+static void can_msg_handler(const can_msg_t *msg);
+
+static enum ARM_STATE alt_1_arm_state = DISARMED;  //this should be ARMED for flight code
+static enum ARM_STATE alt_2_arm_state = DISARMED;
+
+// Memory pool for CAN transmit buffer
+uint8_t tx_pool[100];
+
+int main(int argc, char** argv) {
+    //init functions
+    osc_init();
+    timer0_init();    
+    output_init(); // initialize our outputs
+    uart1_rx_init(9600, _XTAL_FREQ); //initialize the UART module
+    // MCC generated functions
+    ADCC_Initialize(); 
+    FVR_Initialize();
+    
+    // enable global interupts
+    INTCON0bits.GIE = 1;
+    
+    /***********set up CAN**********/
+    
+    // Set up CAN TX
+    TRISC0 = 0;
+    RC0PPS = 0x33;
+
+    // Set up CAN RX
+    TRISC1 = 1;
+    ANSELC1 = 0;
+    CANRXPPS = 0x11;
+    
+    // set up CAN module
+    can_timing_t can_setup;
+    can_generate_timing_params(_XTAL_FREQ, &can_setup);
+
+    can_init(&can_setup, can_msg_handler);
+    // set up CAN tx buffer
+    txb_init(tx_pool, sizeof(tx_pool), can_send, can_send_rdy);
+    
+    
+    uint32_t last_millis = millis();
+    uint8_t on = 0;
```

Comment:
`on` is a pretty enigmatic variable name. It looks like it just deals with the white LED, so it could be named as such. Also `uint8_t` here, while not wrong, may be more readable as `bool`.

Reply:
I realized I can handle this in a way more fun way:
LATC5 = ~LATC5;

Reply:
You want LATC5 ~= PORTC5

LAT is used for writing, PORT for reading. Though in this case it doesn't make a huge difference

Reply:
god this took me to long to fix
ANSEL was set for the pin, so reading from PORTC5 wasn't working.

Reply:
LOL

Reply:
One of us one of us